### DISCUSSION TITLE
Question: how to get a list of values from a queryset?

### DISCUSSION THREAD

**Discussion by bolinocroustibat** [collerek/ormar#222] [Q&A] [answered]
I want to get the list of records ids with a filtered attribute (to be able to pick a random one among them later).

In Django ORM, I would do something like:
`records = Item.objects.filter(attribute=1).values_list('id', flat=True)`

In SQL Alchemy, something like https://stackoverflow.com/questions/24560945/sqlalchemy-get-list-of-ids-from-a-query

How to do that with Ormar? Thanks!

**Comment by collerek** [comment]
Right now there is no way to return the raw data from a query bypassing the model -> sounds like an enhancement idea.

So for now you would have to do `[x.id for x in await Item.objects.filter(attribute=1).all()]` (I assume id is unique here as it looks like pk, otherwise use set comprehension instead of list). SO it's still gonna parse the models which you do not need in this case, but that will work.

Note that if you have other fields optional you can select only id i.e. `Item.objects.filter(attribute=1).fields("id").all()`

**Reply by bolinocroustibat** [reply]
Thanks. And yes, that would be definitely a good enhancement.

**Reply by collerek** [reply]
Created enhancement issue

**Reply by collerek** [reply]
since 0.10.11 you can `records = await Item.objects.filter(attribute=1).values_list('id', flatten=True)`

**Comment by collerek** [comment, marked as answer]
Added in 0.10.11
